base/math.jl

```diff
@@ -349,8 +349,49 @@ julia> ldexp(5., 2)
 20.0
 ```
 """
-ldexp(x::Float64,e::Integer) = ccall((:scalbn,libm),  Float64, (Float64,Int32), x, Int32(e))
-ldexp(x::Float32,e::Integer) = ccall((:scalbnf,libm), Float32, (Float32,Int32), x, Int32(e))
+function ldexp{T<:AbstractFloat}(x::T, e::Integer)
```

Comment:
I guess it assumes:
1. Is a `bitstype`.
2. Is base-2
3. Uses the same layout as the IEEE format (sign | biased exponent | significand )
4. Uses gradual underflow for subnormals